src/sql/parts/notebook/outputs/renderers.ts

```diff
@@ -265,6 +271,88 @@ export function renderLatex(
 	return Promise.resolve(undefined);
 }
 
+/**
+ * Render DataResource as a grid into a host node.
+ *
+ * @params options - The options for rendering.
+ *
+ * @returns A promise which resolves when rendering is complete.
+ */
+export function renderDataResource(
```

Comment:
Can we put this to its own file? I think we'll want to do a lot more here, so would be good to have it's own place where we can put lots of logic